dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -71,84 +93,226 @@ function Catalog(props: ICatalogProps) {
-    pushFilters({
-      ...filters,
-      [type]: filters[type].concat(value),
-    });
+    pushFilters(
+      {
+        ...filters,
+        [type]: filters[type].concat(value),
+      },
+      type,
+    );
   };
   const removeFilter = (type: string, value: string) => {
-    pushFilters({
-      ...filters,
-      [type]: without(filters[type], value),
-    });
+    pushFilters(
+      {
+        ...filters,
+        [type]: without(filters[type], value),
+      },
+      type,
+    );
   };
   const removeFilterFunc = (type: string, value: string) => {
     return () => removeFilter(type, value);
   };
-  const clearAllFilters = () => {
-    pushFilters({});
+  const removeSearchQuery = () => {
+    return () => {
+      dispatch(actions.charts.resetChartsSearch());
+      setCurrentSearchQuery("");
+    };
   };
-  const submitFilters = () => {
-    pushFilters(filters);
+  const clearAllFilters = () => {
+    if (filters[filterNames.REPO]) {
+      dispatch(actions.charts.resetPaginaton());
+    }
+    dispatch(actions.charts.resetChartsSearch());
+    setCurrentSearchQuery("");
+    pushFilters({}, "");
   };
 
-  const allRepos = uniq(charts.map(c => c.attributes.repo.name));
+  const allRepos = uniq(repos.map(c => c.metadata.name));
   const allProviders = uniq(csvs.map(c => c.spec.provider.name));
   const allCategories = uniq(
-    charts
-      .map(c => categoryToReadable(c.attributes.category))
+    categories
+      .map(c => categoryToReadable(c.name))
       .concat(flatten(csvs.map(c => getOperatorCategories(c)))),
   ).sort();
 
+  // We do not currently support app repositories on additional clusters.
+  const supportedCluster = cluster === kubeappsCluster;
+
   useEffect(() => {
-    fetchCharts(cluster, namespace, repo);
-    getCSVs(cluster, namespace);
-  }, [cluster, namespace, repo, fetchCharts, getCSVs]);
+    const newFilters = {};
+    Object.keys(propsFilter).forEach(filter => {
+      newFilters[filter] = propsFilter[filter]?.toString().split(",");
+    });
+    const repo = propsFilter[filterNames.REPO]?.toString();
+    if (repo) {
+      setCurrentRepo(repo);
+    }
+    setFilters({
+      ...initialFilterState(),
+      ...newFilters,
+    });
+  }, [propsFilter]);
+
+  const firstUpdate = useRef(true);
+
+  const namespaceUpdate = useRef({ namespace });
+  useEffect(() => {
+    if (firstUpdate.current || namespaceUpdate.current.namespace !== namespace) {
+      // initial actions or if only the namespace is changing, refresh
+      firstUpdate.current = false;
+      setCurrentSearchQuery("");
+      setCurrentRepo("");
+      setFilters({ ...initialFilterState() });
+      dispatch(actions.charts.resetPaginaton());
+      dispatch(actions.charts.resetChartsSearch());
+      fetchChartsWithPagination(cluster, namespace, currentRepo, "", 1, size, nextPage); // get the first charts
+      getCSVs(cluster, namespace);
+      dispatch(actions.charts.resetPaginaton()); // start from page=1 again
```

Comment:
Why? Won't this mean the next page of results requested will be the first page again? (guessing not, but if you could expand the comment to explain *why* that'd be helpful.

Reply:
Right now I'm not able to recall why the `resetPaginaton()` was useful. I remember I was getting page=2 results even if I changed the filters, but I'm not able to reproduce it right now. Commenting this line, but not deleting it just in case.